R/step-join.R

```diff
@@ -88,7 +88,16 @@ left_join.dtplyr_step <- function(x, y, ..., by = NULL, copy = FALSE, suffix = c
   by <- dtplyr_common_by(by, x, y)
 
   if (join_is_simple(x$vars, y$vars, by)) {
-    step_subset_on(y, x, i = y, on = by)
+    col_order <- unique(c(x$vars, y$vars))
+    out <- step_subset_on(y, x, i = y, on = by)
+
+    step_call(
```

Comment:
I think it's probably time to make `step_col_order()`. Would you mind doing that and using it here and in `pivot_wider()`? 

@markfairbanks should we also use it in `relocate()`?

Reply:
@hadley done. This was actually easier than expected 😄

Reply:
@hadley Yep, I think we should. It appears to be decently faster than using "select". I'll work on a PR.

``` r
library(data.table)

data_size <- 100000
test_df <- 
  data.table(
    x = sample(1:5, data_size, TRUE),
    y = sample(letters[1:5], data_size, TRUE),
    z = sample(LETTERS[1:5], data_size, TRUE)
  )

bench::mark(
  select = test_df[, c("z", "y", "x")],
  setcolorder = setcolorder(copy(test_df), c("z", "y", "x"))[]
)
#> # A tibble: 2 x 6
#>   expression       min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>  <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 select        1.35ms   1.59ms      561.    3.45MB     25.5
#> 2 setcolorder 276.24µs 400.54µs     2024.    1.94MB    119.
```

Reply:
Cool, thanks!